schema/artist/index.js

```diff
@@ -193,6 +195,21 @@ export const ArtistType = new GraphQLObjectType({
           )
         },
       },
+      auctionResults: {
+        type: auctionResultConnection,
+        args: pageable({}),
+        resolve: ({ _id }, options, _request, { rootValue: { auctionLotLoader } }) => {
+          // Convert `after` cursors to page params
+          const { page, size, offset } = parseRelayOptions(options)
+          const diffusionArgs = { page, size, artist_id: _id }
+          return auctionLotLoader(diffusionArgs).then(({ total_count, _embedded }) => {
+            return connectionFromArraySlice(_embedded.items, options, {
```

Comment:
Instead of using a node HAL client, we just hardcoded the route/path. /shrug

Its....sort of unclear if we settled on any specific practice. For instance, Diffusion uses HAL. Convection has a GraphQL implementation which is [soon to be] stitched into Metaphysics.

Should we attempt the same thing with Diffusion? cc @joeyAghion

Reply:
I think we weren't sure about putting GraphQL directly in front of a Postgres DB from a performance standpoint. We need to at the very least heavily restrict sorts. We also didn't think clients would be hitting it directly (those calls would go via MP). But I'm not too familiar with how stitching works so not sure what the upside looks like if we go for GraphQL in Diffusion.

Reply:
> we weren't sure about putting GraphQL directly in front of a Postgres DB from a performance standpoint

What are the performance concerns in the context of putting this close to the DB?

> We also didn't think clients would be hitting it directly (those calls would go via MP).

Clients would still go through MP, but parts of the query that access data that Diffusion is responsible for would get proxied to Diffusion’s theoretical GraphQL endpoint, rather than implementing that schema here and accessing the data via a REST approach.

Reply:
Decisions like these may also need to take into account if the service actually exposes any data that could be represented as a graph. If Diffusion _only_ ever exposes a single lot (without nested data) or a list thereof, then GraphQL maybe doesn’t inherently bring more to the table than HAL, other than being in control of what the lots schema looks like in Diffusion rather than in MP.

Reply:
> Decisions like these may also need to take into account if the service actually exposes any data that could be represented as a graph. If Diffusion only ever exposes a single lot (without nested data) or a list thereof, then GraphQL maybe doesn’t inherently bring more to the table than HAL, other than being in control of what the lots schema looks like in Diffusion rather than in MP.

Very good point. I have to think, my gut says that auction results are an important first-class 'thing' that Diffusion manages, as well as serves. I think serving a single result or list endpoint (accepting various params to filter a list- as it does currently), _may_ be the final stable resting place for Diffusion. Which would imply that we don't need to add GraphQL there.

@joeyAghion do you have any thoughts on potential other endpoints/API offerings that Diffusion may have? That might help inform whether we should explore adding GraphQL there, or not really.

Reply:
I'd be 👍 to supporting Graphql at Diffusion if we merge and deploy the latest stitching work (around Convection) and have a good enough experience that we want it to be the convention. Diffusion's APIs pre-dated that but no reason we can't change them since Metaphysics will be the sole consumer.

Reply:
If stitching is a pattern we are adopting i'm happy to give it a try too.

@alloy The performance concern was lack of control over access patterns if a client can select anything, join anything and sort on anything by default. I'm sure we can restrict that somehow, but that was coupled with the fact we didn't see browser based clients needing to hit the API, so we weren't convinced of upside given the maturity of GraphQL. The first client hitting the API is Gravity (@joeyAghion), and using Hyperclient there is pretty convenient.